### PR TITLE
Added sklearn example

### PR DESCRIPTION
**Pull request by kamil-kaczmarek**

### CODE REVIEW

**Comment by kamil-kaczmarek** [comment]
https://ui.neptune.ai/o/shared/org/sklearn-integration/experiments?viewId=standard-view

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 1]
```diff
@@ -0,0 +1,2048 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "### Step 1: Import libraries"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:310]: Again, I'd combine those into one sklearn-related block

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 2]
```diff
@@ -0,0 +1,2048 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "AW4MM4qj5ErX",
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "### Step 3: Create Neptune experiment and log regressor summary"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:367]: Again, I'd split it into two, add the part related to token, and the part click on a link to see.

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 3]
```diff
@@ -0,0 +1,2048 @@
...
+   "source": [
+    "import neptune\n",
+    "from neptunecontrib.monitoring.sklearn import log_classifier_summary\n",
+    "\n",
+    "neptune.init('shared/sklearn-integration',\n",
+    "             api_token='ANONYMOUS')\n",
+    "exp = neptune.create_experiment(name='classification-example',\n",
+    "                                tags=['GradientBoostingClassifier', 'classification'])\n",
+    "\n",
+    "log_classifier_summary(gbc, X_train, X_test, y_train, y_test, experiment=exp)"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:395]: I think we should add a comment with what was logged and that the test prediction .csv can be huge

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 4]
```diff
@@ -0,0 +1,2048 @@
...
+   "source": [
+    "import neptune\n",
+    "from neptunecontrib.monitoring.sklearn import log_regressor_summary\n",
+    "\n",
+    "neptune.init('shared/sklearn-integration',\n",
+    "             api_token='ANONYMOUS')\n",
+    "exp = neptune.create_experiment(name='regression-example',\n",
+    "                                tags=['RandomForestRegressor', 'regression'])\n",
+    "\n",
+    "log_regressor_summary(rfr, X_train, X_test, y_train, y_test, experiment=exp)"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:188]: I think we should add a comment with what was logged and that the test prediction .csv can be huge

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 5]
```diff
@@ -0,0 +1,2048 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "QXGSKS1u0n8x",
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "# Scikit-learn + Neptune"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:13]: I think we are missing some parts where:

- we show that you can disable certain options 
- we show you can just use a specific function to log something. 

I'd add those to give a good view of this integration.

File: integrations/sklearn/tests/Neptune-Scikit-learn_upgraded_libs.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+# check parameters
+assert set(exp.get_properties().keys()) == set(km.get_params().keys()), '{} parameters do not match'.format(exp)
+
+## Step 4: Stop Neptune experiment after logging summary
+
+exp.stop()
+
+## Explore Results
+
+# tests
```
Comment (jakubczakon) [inline integrations/sklearn/tests/Neptune-Scikit-learn_upgraded_libs.py:171]: I think I'd drop this test potentially. 
Remember that we are testing the example (and whether things work with newer versions and not the library itself -> this should be done in contrib

**Review by jakubczakon — CHANGES_REQUESTED** [review]

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 7]
```diff
@@ -0,0 +1,2048 @@
...
+    "colab": {},
+    "colab_type": "code",
+    "id": "DO7yawFVxVIt",
+    "tags": [
+     "installation"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "! pip install scikit-learn==0.23.2 neptune-client==0.4.129 neptune-contrib[monitoring]==0.25.0"
```
[the remark this answers is not in the document]
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:64]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 8]
```diff
@@ -0,0 +1,2048 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "### Step 1: Import libraries"
```
[the remark this answers is not in the document]
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:103]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 9]
```diff
@@ -0,0 +1,2048 @@
...
+    "colab_type": "code",
+    "id": "GvDSBSrOx-R4",
+    "outputId": "667d1b2f-cdef-44f0-f505-832205dea223",
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "import neptune\n",
```
[the remark this answers is not in the document]
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:180]: done

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 10]
```diff
@@ -0,0 +1,2048 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment"
+    ]
+   },
+   "source": [
+    "Neptune helps you keep track of your machine learning experiments and if you are using scikit-learn you can add tracking very easily.\n",
+    "\n",
+    "This quickstart will show you how to log scikit-learn regression, classification and kmeans clustering summary to Neptune using single function."
```
[the remark this answers is not in the document]
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:26]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 11]
```diff
@@ -0,0 +1,2048 @@
...
+    "assert set(exp.get_properties().keys()) == set(rfr.get_params().keys()), '{} parameters do not match'.format(exp)\n",
+    "\n",
+    "# check estimator\n",
+    "import joblib\n",
+    "import os\n",
+    "import tempfile\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "with tempfile.TemporaryDirectory() as d:\n",
```
[the remark this answers is not in the document]
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:217]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 1 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:310]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 2 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:367]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 3 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:395]: It will be 1k lines. Added comment what is logged

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 4 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:188]: It will be 1k lines.

Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:188]: Added comment what was logged

File: integrations/sklearn/tests/Neptune-Scikit-learn_upgraded_libs.py (hunk 6 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/tests/Neptune-Scikit-learn_upgraded_libs.py:171]: dropped.

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 5 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:13]: done

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 12]
```diff
@@ -0,0 +1,2401 @@
...
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.datasets import load_boston\n",
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "rfr = RandomForestRegressor(n_estimators=70, max_depth=7, min_samples_split=3)\n",
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:124]: I think we should show that you can/should track those parameters.

so I'd simply create a dict (including seed and test size perhaps) here and later pass it to create you know

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 13]
```diff
@@ -0,0 +1,2401 @@
...
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "neptune.create_experiment(name='regression-example',\n",
+    "                          tags=['RandomForestRegressor', 'regression'])"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:214]: I'd add params here for sure

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 14]
```diff
@@ -0,0 +1,2401 @@
...
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "tests"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# tests\n",
+    "exp = neptune.get_experiment()\n",
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:294]: I think it is safer to divide tests into two cells and have `neptune.stop()` right after this.
It will ensure that all of those objects are logged.

File: integrations/sklearn/Neptune-Scikit-learn.ipynb [hunk 15]
```diff
@@ -0,0 +1,2401 @@
...
+    "colab_type": "code",
+    "id": "GvDSBSrOx-R4",
+    "outputId": "667d1b2f-cdef-44f0-f505-832205dea223",
+    "tags": [
+     "code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "exp.stop()"
```
Comment (jakubczakon) [inline integrations/sklearn/Neptune-Scikit-learn.ipynb:335]: I'd change this to netpune.stop() as the exp is actually only in the tests.

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 12 above)
Comment (jakubczakon) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:124]: The same goes for other sections

**Review by jakubczakon — CHANGES_REQUESTED** [review]

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 13 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:214]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 12 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:124]: ok

File: integrations/sklearn/Neptune-Scikit-learn.ipynb (hunk 14 above)
Comment (kamil-kaczmarek) [reply integrations/sklearn/Neptune-Scikit-learn.ipynb:294]: done

**Review by jakubczakon — APPROVED** [review]
